ethereum/remix
# comeoutieDque (#2)

* move debugger logic from remix-ide to remix-debug

* fix requires

* fix requires

* remove unneded reset

* make linter happy

* add missing param

* fix requires

* move execution context out of original debugger

* move setManagers to debugger; cleanup

* cleanup

* move initializations of execution context

* move providers methods out of debugger

* make linter happy

* comment test for now (needs to be rewriten with new api)

* remove unneed console log

* add callback to state manager checkpoint call

* add missing send method

* add websocket support for remix-sim

* make linter happy

* Add tests, research findImportsSync

* make linter happy

* add missing return

* add missing stringify

* move non prod dependencies to devDependencies

* remove unused packages

* update readme; remove unused requires

* Update tests.sol.js

warnings

* rename filename to fix uppercase issue in the require

* try to implement getFile

* resolve via callback

* callback & compile

* Initialize remix solidity resolve repository

* tests for remix-resolve

* add tests

* pass tests, add I/O spec

* update path of test contracts

* add github import tests. ## First commit from Etheratom office ##

* Add test for successful compilation test

* Check compilation with ast

* Add tests, research findImportsSync

* remove unused packages

* update readme; remove unused requires

* try to implement getFile

* resolve via callback

* callback & compile

* add circleci config

* update tests

* add remix-resolve to lerna

* Try typescript

* add dist to gitignore

* add dist to gitignore

* add dist to gitignore

* Try to implement resolve function in TS

* first implementation of resolve with custom handler

* remove useless code; update test; update circleCI script

* try async ipfs request

* use array filter instead forEach

* successful async/await implementation of IPFS handler; update suggestions

* fix txFormat tests

* test

* init providers only after already listening to events

* add offsetToLineColumnConverter to remix-lib

* re-add offset component; add missing vmdebugger start

* update test script

* add basic cmd line debugger

* improve displaying current step

* add resolveToReducedTrace; clean up; improve output

* update lib; remove various console logs

* don't print source automatically; has to be required instead

* display errors

* add more events

* remove unneeded console logs; prevent crash when step is out of bounds

* add callbacks so each step of the debugger is only called when it's ready

* add loadCompilationData method

* improve api by adding common methods

* add jumpto and get trace length steps

* fix breakpoint event; add method to trigger souce update

* add methods needed for solidity code navigation

* cleanup

* remove uneeded file; remove decoupling

* make linter happy

* Work on #1013

* Work on multiple directories

* less semicolons

* Less semicolons

* Expect full contract name in test run

* unit testing fix for PR #1014

* Use gatherContractsFrom to collect contracts in fs.walkSync

* split index.js into multiple files, reuse functions

* use filepath again

* fix conflicts

* less semicolons

* Better indentation

* Less semicolons

* fix conflicts

* Use gatherContractsFrom to collect contracts in fs.walkSync

* use filepath again

* fix conflicts

* fix conflicts

* fix conflicts

* path 0.5.0 updates for remix-tests testing

* always append remix-tests:sol reference

* Add typescript tests

* write basic AppManager tests

* Use remix-plugin module API

* access module through AppManager

* Add IPFS tests

* Add tests with & without AppManager

* add github import handler

* Add github import handler & tests

* handle https imports

* Add http call handle, add tests, add more typings

* put package versions in line with npm

* fix teste

* remove direct dependency of remix-plugin

* Updated Readme

* rename package to remix-url-resolver

* update cirlce-ci tests

* lerna config

* fix remix-url-resolver package.json

* Update release-process.md

* refactor server code into a module

* add command line options

* create rdb bin

* make linter happy

* remove bin config from package.json

* Update readme

* Update package.json

* fix tests

* improve userdoc for sender account

* Update README.md

* Add module.d.ts

* add remix-tests tsconfig

* update tsconfig to generate declaration

* Try to create a typescript version of  remix-tests module

* Allow calls to runRaw to specify the timestamp.

This is useful when, for example, running a recorded .json
file for cases that rely on a specific time between transactions.

If no timestamp is passed as part of the tx argument to
runRaw, behavior does not change.

* Reverting changes to package.json

* Fix indent error in txRunner

* migrated to ts [WIP]

* add typescript tests

* test migrated to typescript

* log transaction error

* make tests pass in typescript

* remove dist directory from git

* add more typings to src/testRunner.ts

* passing all tests

* compiler, deployer, logger migrated to ts

* rename fs.ts

* use types/web3

* Add more typings

* Update contract to sol v0.5.0, add more typings, make runTestFiles work

* fix: storageRangeAt doesn't always take a callback

* Updated regex in Memory Debugger to exclude replacing space with '?'

* fix test

* pass safe_math_tests

* Add types file

* Add more fixes, arrow functions

* fix ES6 mistakes

* replace var with const in tests

* ethereum/populus is no longer maintained. Use ours.

* Fix typo

* run remix-lib tests with v0.5.4+commit.9549d8ff

* ignore linting rule

* fix: handle bignumber conversion on hash without prefix

* add remix-tests tsconfig

* Try to create a typescript version of  remix-tests module

* import mocha in testRunner to remove warnings

* constantinople support

* forked js-vm(see ethereumjs-vm/6192663ac68e03aac73d5ddcb7697001af2f37fb)

* Update execution-context.js

* add remix-tests tsconfig

* Try to create a typescript version of  remix-tests module

* migrated to ts [WIP]

* add typescript tests

* test migrated to typescript

* log transaction error

* make tests pass in typescript

* remove dist directory from git

* add more typings to src/testRunner.ts

* passing all tests

* compiler, deployer, logger migrated to ts

* rename fs.ts

* use types/web3

* Add more typings

* Update contract to sol v0.5.0, add more typings, make runTestFiles work

* pass safe_math_tests

* Add types file

* Add more fixes, arrow functions

* fix ES6 mistakes

* replace var with const in tests

* add remix-tests tsconfig

* import mocha in testRunner to remove warnings

* tsconfig minor changes

* bump versions

* bump versions

* Update package.json

* Update package.json

* bump version

* Update README.md

* fix runTestSources

* fix require

* Add types to eachOfLimit callback

* bump ethereumjs-vm version

* bump lerna

* incr root
